### PR TITLE
Implement Dynamic country change

### PR DESCRIPTION
**Pull request by DivyaniPatel96**
Add country API and make dynamic

### CODE REVIEW

File: app/assets/translations/appOfflineData/apiConstants.ts [hunk 1]
```diff
@@ -70,8 +72,19 @@ export const appConfig = {
   checkUpdate: 'check-update',
   faqs: 'faqs',
   archive: 'archive',
+  countryGroups: 'country-groups',
 }
 export const finalUrl = (apiEndpoint: string, selectedCountry: number | undefined, selectedLang: string): any => {
+  if (apiEndpoint == appConfig.countryGroups) {
```
Comment (Akhror) [inline app/assets/translations/appOfflineData/apiConstants.ts:78]: Should we use switch statement instead? For example:

```
export const finalUrl = (
  apiEndpoint: string,
  selectedCountry: number | undefined,
  selectedLang: string
): string => {
  const baseUrl = `${apiUrlDevelop}/${apiEndpoint}`;

  switch (apiEndpoint) {
    case appConfig.countryGroups:
      return `${baseUrl}/${buildFor === String(buildForFoleja) ? folejaName : bebboName}`;
    case appConfig.sponsors:
      return `${baseUrl}/${selectedCountry}`;
    case appConfig.taxonomies:
      return `${baseUrl}/${selectedLang}/all`;
    case appConfig.checkUpdate:
      return `${baseUrl}/${selectedCountry}`;
    default:
      return `${baseUrl}/${selectedLang}`;
  }
};
```

File: app/assets/translations/appOfflineData/bebbo/countryData.ts [hunk 2]
```diff
@@ -0,0 +1,541 @@
+export const countryData: any = {
+    "en": [
+        {
+            "CountryID": "6",
```
Comment (Akhror) [inline app/assets/translations/appOfflineData/bebbo/countryData.ts:4]: Be consistent throughout the file. Follow the pattern on the left attributes. For example:
1. Use camel case: countryId, countryEmail etc.
or 
2. Use underscore with all lowercase: country_id, country_email etc.

File: app/redux/sagaMiddleware/saga.ts [hunk 3]
```diff
@@ -50,6 +51,10 @@ function* onApiSuccess(response: AxiosResponse<any>, prevPage: string, dispatch:
   else if (prevPage == 'CountryLanguageSelection') {
     yield call(onSponsorApiSuccess, response, dispatch, navigation, languageCode, prevPage)
   }
+  else if (prevPage == '') {
+    console.log('Dsxucgfsid')
```
Comment (Akhror) [inline app/redux/sagaMiddleware/saga.ts:55]: Remove log or imrpove.

File: app/redux/reducers/utilsSlice.ts [hunk 4]
```diff
@@ -196,6 +198,10 @@ export const utilsSlice = createSlice({
       action: PayloadAction<any>,):any=>{
           state.userIsOnboarded = action.payload;
     },
+    setuserIsFirstTime:(  state,
+      action: PayloadAction<any>,):any=>{
+          state.userIsFirstTime = action.payload;
+    },
```
Comment (Akhror) [inline app/redux/reducers/utilsSlice.ts:204]: Improve the format.

File: app/navigation/AppNavigation.tsx [hunk 5]
```diff
@@ -794,9 +848,22 @@ export default (): any => {
...
+  // if (dataStatus !== 200) {
+  //   return <ActivityIndicator size="large" color="#0000ff" />;
+  // }
+  // if (apiStatus === 'loading') {
+  //   return <ActivityIndicator size="large" color="#0000ff" />;
+  // }
 
+  // if (apiStatus === 'failed') {
+  //   return <Text>Error: {apiError}</Text>;
+  // }
```
Comment (Akhror) [inline app/navigation/AppNavigation.tsx:863]: Remove commented code.

File: app/navigation/AppNavigation.tsx [hunk 6]
```diff
@@ -147,9 +161,49 @@ export default (): any => {
...
+  //   }
+  //   setCountries();
+  //   console.log('loading is', loading);
+   
+  // }, [userIsFirstTime, dispatch]);
+
+  useEffect(() => {
+    console.log('userIsFirstTime is',userIsFirstTime)
+    if(!userIsFirstTime){
+    dispatch(fetchAPI(apiJsonData, '', dispatch, navigationRef.current, languageCode, activeChild, apiJsonData, netInfo.isConnected))
```
Comment (Akhror) [inline app/navigation/AppNavigation.tsx:201]: Indentation

File: app/navigation/AppNavigation.tsx [hunk 7]
```diff
@@ -147,9 +161,49 @@ export default (): any => {
...
+  //       console.log(e.message);
+  //     } finally {
+  //       console.log('afterwdwdwdwdwdwd',loading)
+  //       setLoading(false);
+  //     }
+  //   }
+  //   setCountries();
+  //   console.log('loading is', loading);
+   
+  // }, [userIsFirstTime, dispatch]);
```
Comment (Akhror) [inline app/navigation/AppNavigation.tsx:196]: Remove commented code.

File: app/navigation/AppNavigation.tsx [hunk 8]
```diff
@@ -147,9 +161,49 @@ export default (): any => {
       resetURL();
     }
   }
+  const apiJsonData = [
+    {
+      apiEndpoint: appConfig.countryGroups,
+      method: 'get',
+      postdata: {},
+      saveinDB: true,
```
Comment (Akhror) [inline app/navigation/AppNavigation.tsx:169]: Use consistent naming conventions (e.g. camelCase) for left attributes.

File: app/localization/initI18next.ts [hunk 9]
```diff
@@ -1,90 +1,90 @@
...
+  detect: (callback: (lng: string) => void) => {
+    AsyncStorage.getItem('APP_LANG', (err, lng) => {
+      // Error fetching stored data or no language was stored
+      if (err || !lng) {
+        
+        const bestLng = RNLocalize.findBestLanguageTag(AVALAILABLE_LANG_CODES);
+       const langCodeNew = findLangCode(bestLng?.languageTag);
+       const lang2 = langCodeNew ?langCodeNew : localization[localization.length-1]?.languages[0]?.locale;
+       const country = localization.find((x:any) => x.languages.some((item:any) => item.locale === lang2));
+      const language = localization.reduce((prev: any, product: any) => prev || product.languages.find((item:any) => item.locale === lang2), undefined);
```
Comment (Akhror) [inline app/localization/initI18next.ts:54]: Improve format of the newly added code.

File: app/localization/initI18next.ts [hunk 10]
```diff
@@ -1,90 +1,90 @@
...
+import { store } from '../../App';
+import { onLocalizationSelect } from '../redux/reducers/localizationSlice';
+import { localization, AVAILABLE_LANGUAGES } from '@dynamicImportsClass/dynamicImports';
 
-  const findLangCode = (languageTag: string | undefined):any => {
-    const obj = localisationnew.reduce((prev, product):any => prev || product.languages.find((item:any) => item.luxonLocale === languageTag && item.locale != 'RSen'), undefined);
-    const obj2 = obj ? obj.locale : obj;
-    return obj2;
+console.log("AVAILABLE_LANGUAGES--",AVAILABLE_LANGUAGES);
+const localisationnew = [...localization];
```
Comment (Akhror) [inline app/localization/initI18next.ts:14]: localizationNew

File: app/database/schema/CountrySchema.ts [hunk 11]
```diff
@@ -0,0 +1,69 @@
...
+export type Partner = {
+  url: string;
+  name: string;
+  alt: string;
+};
+
+export type Country = {
+  CountryID: string;
+  name: string;
+  country_email: string;
```
Comment (Akhror) [inline app/database/schema/CountrySchema.ts:22]: Can we stick to camel casing throughout the file?

File: app/components/homeScreen/DailyReads.tsx [hunk 12]
```diff
@@ -205,103 +161,103 @@ const DailyReads = (): any => {
...
       if (activeChild.isPremature === 'true') {
         filteredArticles = ArticlesData.filter((article: any) => article.premature === 1).sort((a: any, b: any) => new Date(b.created_at) - new Date(a.created_at));
         ArticlesData = filteredArticles;
       }
-      const articleCategoryArrayNew = articleCategoryArray.filter((i: any) => ArticlesData.find((f: any) => f.category === i))
-      const activityCategoryArrayNew = activityCategoryArray.filter((i: any) => ActivitiesData.find((f: any) => f.activity_category === i.id))
+      const articleListData: any = [];
+      const activityListData: any = [];
+      const articleCategoryArrayNew = articleCategoryArray.filter((i: any) => ArticlesData.find((f: any) => f.category === i));
+      const activityCategoryArrayNew = activityCategoryArray.filter((i: any) => ActivitiesData.find((f: any) => f.activity_category === i.id));
```
Comment (Akhror) [inline app/components/homeScreen/DailyReads.tsx:173]: I would name these as follows:
```
articleList
activityList
articleCategoryArrayFiltered
activityCategoryArrayFiltered
```

File: app/screens/home/SettingScreen.tsx [hunk 13]
```diff
@@ -664,70 +706,96 @@ const SettingScreen = (props: any): any => {
...
-            const importResponse = await backup.importFromFile(oldChildrenData, props.navigation, genders, dispatch, childAge, languageCode);
-            console.log(importResponse, "..importResponse");
+            try {
+              console.log("..importResponse");
+              await userRealmCommon.openRealm();
+              await userRealmCommon.deleteAllAtOnce();
+              const importResponse = await backup.importFromFile(oldChildrenData, props.navigation, genders, dispatch, childAge, languageCode);
+              
+            } catch (error) {
+              console.log(error, "..importResponse error");
```
Comment (Akhror) [inline app/screens/home/SettingScreen.tsx:790]: Use console.error

File: app/screens/home/SettingScreen.tsx [hunk 14]
```diff
@@ -664,70 +706,96 @@ const SettingScreen = (props: any): any => {
...
             console.log('importedJsonData data is',importedJsonData)
             oldChildrenData = importedJsonData;
-            await RNFS.writeFile(tempRealmFile, JSON.stringify(importedJsonData), "utf8");
+            console.log("oldChildrenData-if", oldChildrenData);
+            try {
+              //console.log("tempRealmFile", tempRealmFile);
+              
+              await RNFS.writeFile(tempRealmFile, JSON.stringify(importedJsonData), "utf8");
+            } catch (error) {
+              //console.log("error catch", error);
```
Comment (Akhror) [inline app/screens/home/SettingScreen.tsx:761]: Handle the exception

File: app/screens/home/SettingScreen.tsx [hunk 15]
```diff
@@ -598,16 +617,39 @@ const SettingScreen = (props: any): any => {
...
-    navigation.dispatch(DrawerActions.closeDrawer());
-    const selectedCountry: any = localization.find(
-      (country: any) => country.countryId === countryId,
-    );
-    setCountry(selectedCountry);
-    const selectedLanguage: any = selectedCountry.languages.find(
-      (language: any) => language.languageCode === languageCode,
-    );
-    setlanguage(selectedLanguage);
+    console.log('Selected country is...', allCountries);
```
Comment (Akhror) [inline app/screens/home/SettingScreen.tsx:634]: All countries

File: app/screens/home/SettingScreen.tsx [hunk 16]
```diff
@@ -598,16 +617,39 @@ const SettingScreen = (props: any): any => {
...
+  //   const selectedCountry: any = allCountries.find(
+  //     (country: any) => country.CountryID === countryId,
+  //   );
+  //   setCountry(selectedCountry);
+  //   const selectedLanguage: any = selectedCountry.languages.find(
+  //     (language: any) => language.languageCode === languageCode,
+  //   );
+  //   setlanguage(selectedLanguage);
+  //   toggleSwitch();
+  // }, []);
```
Comment (Akhror) [inline app/screens/home/SettingScreen.tsx:632]: Remove commented code if not necessary.

File: app/screens/home/SettingScreen.tsx [hunk 17]
```diff
@@ -177,11 +184,21 @@ const SettingScreen = (props: any): any => {
...
+    
     return AesCrypto.encrypt(text, key, encryptionsIVKey, 'aes-256-cbc').then((cipher: any) => ({
       cipher
     }));
   }
   const decryptData = (text: string, key: any): any => {
+    console.log("decryptData", text);
+    console.log("decryptData key", key);
+    console.log(
+      "decryptData IV key"
```
Comment (Akhror) [inline app/screens/home/SettingScreen.tsx:198]: Modify log text to "Encryptions IV Key"

File: app/screens/home/SettingScreen.tsx [hunk 18]
```diff
@@ -149,6 +152,10 @@ const SettingScreen = (props: any): any => {
   const countryId = useAppSelector(
     (state: any) => state.selectedCountry.countryId,
   );
+  const allCountries = useAppSelector(
+    (state: any) =>
+      state.selectedCountry.countries != '' ? JSON.parse(state.selectedCountry.countries) : [],
+  );
```
Comment (Akhror) [inline app/screens/home/SettingScreen.tsx:158]: Should we add try/catch to handle parsing errors?

const allCountries = useAppSelector(
 (state: any) => {
  try {
    state.selectedCountry.countries !== '' ? JSON.parse(state.selectedCountry.countries) : [];
  } catch (error) {
    console.error('Failed to parse countries JSON:', error);
    return [];
  }
});

File: app/screens/home/SettingScreen.tsx [hunk 19]
```diff
@@ -664,70 +706,96 @@ const SettingScreen = (props: any): any => {
       if (oldChildrenData.length > 0) {
         await userRealmCommon.openRealm();
         await userRealmCommon.deleteAllAtOnce();
+        try {
         console.log("oldchildrenresponse",oldChildrenData)
         const importResponse = await backup.importFromFile(oldChildrenData, props.navigation, genders, dispatch, childAge, languageCode);
+       
         console.log(importResponse, "..importResponse");
+        } catch (error) {
+          console.log(error, "..importResponse error");
```
Comment (Akhror) [inline app/screens/home/SettingScreen.tsx:715]: Use 
`console.error('importResponse error', error);`

File: app/components/shared/ButtonGlobal.tsx [hunk 20]
```diff
@@ -49,13 +49,23 @@ min-height:40px;
...
+  color: ${(props:any):any => props.theme?.colors?.PRIMARY_TEXTCOLOR};
   font-family: 'roboto-bold';
   font-size:16px;
   line-height:20px;
   
 `;
 
+export const ButtonErrorText = styled.Text`
+  text-align: center;
+  color: #00000;
```
Comment (mhdosman) [inline app/components/shared/ButtonGlobal.tsx:61]: please check the color: color: #00000; do you mean: color: #000000;

File: app/services/commonApiService.ts [hunk 21]
```diff
@@ -106,6 +99,73 @@ export const onSponsorApiSuccess = async (response: any, dispatch: any, navigati
...
+   
+    if(response.data && response.data.status && response.data.status == 200)
+    {
+      console.log('type of sponser data for country is', typeof response.data.data)
+      console.log('sponser data for country is',  response.data.data)
+      console.log('sponser data for country issdsd',  response.data?.data[0])
+      dispatch(setuserIsFirstTime(true));
+      const allDatatoStore = await getAllDataToStore(languageCode,dispatch,prevPage);
+      console.log("allDatatoStore ",prevPage,"--",allDatatoStore);
+      //navigation.push('Localization');
```
Comment (mhdosman) [inline app/services/commonApiService.ts:115]: Should we keep this code commented? If the code is finalized, you can remove the commented sections

File: app/services/commonApiService.ts [hunk 22]
```diff
@@ -56,9 +56,7 @@ export const updateIncrementalSyncDT = async(response: any, dispatch: any, _navi
   const articleresp = response.find((y:any)=>y.apiEndpoint == appConfig.articles);
```
Comment (mhdosman) [inline app/services/commonApiService.ts:56]: “Can we improve the variable name ‘artilceresp’? Also, can we adhere to a single formatting scheme?”

File: app/assets/translations/appOfflineData/bebbo/countryData.ts (hunk 2 above)
Comment (DivyaniPatel96) [reply app/assets/translations/appOfflineData/bebbo/countryData.ts:4]: @Akhror this data we got from CMS

File: app/navigation/AppNavigation.tsx (hunk 8 above)
Comment (DivyaniPatel96) [reply app/navigation/AppNavigation.tsx:169]: If we change this then need to change for common method and for all API call.

File: app/screens/localization/CountryLanguageConfirmation.tsx [hunk 23]
```diff
@@ -348,64 +329,70 @@ const CountryLanguageConfirmation = ({ route }: Props): any => {
...
-              <Text style={styles.welcomeText}>{t('welcomeText')}</Text>
-              <Icon name="ic_country" size={100} color="#00AEEF" />
-              <OnboardingshiftHead>
-                <Heading2Centerw>{t('countryLangSelection')}</Heading2Centerw>
-              </OnboardingshiftHead>
-              <Heading4Centerr>{t('checkonce')}</Heading4Centerr>
-            </OnboardingconfirmationHead>
+        <OnboardingContainer>
+
+          <OnboardingconfirmationHead>
```
Comment (Akhror) [inline app/screens/localization/CountryLanguageConfirmation.tsx:343]: Rename OnboardingconfirmationHead to OnboardingConfirmationHead

File: app/screens/localization/CountryLanguageConfirmation.tsx [hunk 24]
```diff
@@ -348,64 +329,70 @@ const CountryLanguageConfirmation = ({ route }: Props): any => {
...
+            <Text style={styles.welcomeText}>{t('welcomeText')}</Text>
+            <Icon name="ic_country" size={100} color="#00AEEF" />
+            <OnboardingshiftHead>
```
Comment (Akhror) [inline app/screens/localization/CountryLanguageConfirmation.tsx:346]: Rename OnboardingshiftHead to OnboardingShiftHead

File: app/screens/localization/CountryLanguageConfirmation.tsx [hunk 25]
```diff
@@ -348,64 +329,70 @@ const CountryLanguageConfirmation = ({ route }: Props): any => {
...
+            </OnboardingshiftHead>
+            <Heading4Centerr>{t('checkonce')}</Heading4Centerr>
+          </OnboardingconfirmationHead>
+
+
+          <LocalizationContainer>
+            <LocalizationRow>
+
+              <LocalizationCol onPress={(): any => navigation.navigate('CountrySelection', { country: countryData, language: newLanguage })}>
+                <LocalizationcontentHead>
```
Comment (Akhror) [inline app/screens/localization/CountryLanguageConfirmation.tsx:357]: Rename LocalizationcontentHead to LocalizationContentHead

File: app/screens/localization/CountryLanguageConfirmation.tsx [hunk 26]
```diff
@@ -348,64 +329,70 @@ const CountryLanguageConfirmation = ({ route }: Props): any => {
...
+            <Text style={styles.welcomeText}>{t('welcomeText')}</Text>
+            <Icon name="ic_country" size={100} color="#00AEEF" />
+            <OnboardingshiftHead>
+              <Heading2Centerw>{t('countryLangSelection')}</Heading2Centerw>
+            </OnboardingshiftHead>
+            <Heading4Centerr>{t('checkonce')}</Heading4Centerr>
+          </OnboardingconfirmationHead>
+
+
+          <LocalizationContainer>
```
Comment (Akhror) [inline app/screens/localization/CountryLanguageConfirmation.tsx:353]: Please format the method. Remove extra lines, add space etc.

File: app/screens/localization/CountryLanguageConfirmation.tsx [hunk 27]
```diff
@@ -348,64 +329,70 @@ const CountryLanguageConfirmation = ({ route }: Props): any => {
...
+            <LocalizationRow>
+
+              <LocalizationCol onPress={(): any => navigation.navigate('CountrySelection', { country: countryData, language: newLanguage })}>
+                <LocalizationcontentHead>
+                  <Heading3Regular>{t('country')}</Heading3Regular>
+                </LocalizationcontentHead>
+                <LocalizationcontentResult>
```
Comment (Akhror) [inline app/screens/localization/CountryLanguageConfirmation.tsx:360]: Rename LocalizationcontentResult to LocalizationContentResult

File: app/screens/localization/CountryLanguageConfirmation.tsx [hunk 28]
```diff
@@ -348,64 +329,70 @@ const CountryLanguageConfirmation = ({ route }: Props): any => {
...
+                  <Heading3>{countryData?.name}</Heading3>
+                </LocalizationcontentResult>
+              </LocalizationCol>
 
+              <LocalizationWithoutBorderCol onPress={(): any => popAndPush(navigation, 'LanguageSelection', { country: countryData, language: newLanguage, luxonlocale: luxonLanLocale != undefined ? luxonLanLocale : null, deviceLanCode: deviceLanCode != undefined ? deviceLanCode : null, isFromCountry: true })}>
```
Comment (Akhror) [inline app/screens/localization/CountryLanguageConfirmation.tsx:365]: Let's simplify:

```
<LocalizationWithoutBorderCol onPress={() =>
          popAndPush(navigation, 'LanguageSelection', {
            country: countryData,
            language: newLanguage,
            luxonLocale: luxonLanLocale ?? null,
            deviceLanCode: deviceLanCode ?? null,
            isFromCountry: true })}>
```

File: app/navigation/AppNavigation.tsx (hunk 8 above)
Comment (Akhror) [reply app/navigation/AppNavigation.tsx:169]: Ah it is fine then.

File: app/assets/translations/appOfflineData/bebbo/countryData.ts (hunk 2 above)
Comment (Akhror) [reply app/assets/translations/appOfflineData/bebbo/countryData.ts:4]: Ah Ok then.

File: app/services/Utils.ts [hunk 29]
```diff
@@ -515,11 +521,12 @@ export const getChild = async (child: any, genders: any): Promise<any> => {
     try {
         const checkPhotoUri = child.photoUri!=null? CHILDREN_PATH +child.photoUri : CHILDREN_PATH;
         const photoUri  = await RNFS.exists(String(checkPhotoUri));     
-        const childmeasures  = await formatImportedMeasures(child.measures)
+        const childmeasures:any  = await formatImportedMeasures(child.measures)
         console.log("reminders output---", childmeasures);
         const childreminders = await formatImportedReminders(child.reminders)
         console.log("reminders output---", childreminders);
         const isPremature: any = child.isPremature == "true" ? "true" : "false";
+        const autoChild: any = child.autoChild == "true" ? "true" : "false";
```
Comment (Akhror) [inline app/services/Utils.ts:529]: You can simplify:

`const autoChild: boolean = child.autoChild === "true";`

**Review by Akhror — APPROVED** [review]

**Pull request merged**